archfx_cloud/scripts/query_factory_map.py

```diff
@@ -0,0 +1,252 @@
+"""
+    ADT-287  Example code for creating a full factory map for flex.
+"""
+import os
+from cryptography.fernet import Fernet
+from typing import Dict, List
+import pickle
+import pandas as pd
+from archfx_cloud.api.connection import Api
+
+DEBUG = False
+
+
+class QueryFactoryMap:
+    def __init__(self, email: str):
+        self._email = email
+        self.token_filename = ".cloud_api_token"
+        self.__initialize_connection()
+        self._sites = {}
+        self._tree = {}
+        self._area = {}
+        self._line = {}
+        self._devices = {}
+
+    def __initialize_connection(self):
+        # api = Api('https://arch.archfx.io')
+        api = Api("https://flex.archfx.io")
+
+        passwd = self.get_password()
+        ok = api.login(email=self._email, password=passwd)
+        if ok:
+            print("logged in successfully!")
+            self.api = api
+            # api.logout()
+
+    # ----------------------------------------------------------------------------------------------------------------
+    # deal with secrets
+    def read_enc_passwd(self):
+        with open(".archfx_enc_passwd", "rb") as reader:
+            return reader.read()
+
+
+    def get_password(self) -> str:
+        key = self.read_token_file()
+        f = Fernet(key)
+        token = self.read_enc_passwd()
+        passwd = f.decrypt(token)
+        passwd = passwd.decode()  # convert to string
+        assert isinstance(passwd, str)
+        return passwd
+
+
+    def read_token_file(self):
+        filename = os.path.abspath(self.token_filename)
+        print(f"filename: {filename}")
+
+        with open(filename, "rb") as reader:
+            key = reader.read()
+            return key
+
+
+    # ----------------------------------------------------------------------------------------------------------------
+    # factory_map generating code
+    def create_factory_map(self):
+        org_name_list = self.query_all_orgs()
+        self.query_all_sites(org_name_list)
+        self.query_all_areas()
+
+        self.query_all_lines()
+        self.save_result_so_far()
+
+        self.query_all_devices()
+        self.save_result_so_far()
+
+
+    def query_all_orgs(self) -> List[str]:
+        result_dict = self.api.org.get()
+        assert isinstance(result_dict, dict)
+        if "results" not in result_dict:
+            return None
+        results = result_dict["results"]
+        org_name_list = []
+        for n, result in enumerate(results):
+            org_name = result.get("slug")
+            if DEBUG:
+                if org_name != "flex-smt":
+                    continue
+            result.pop("avatar")
+            if "thumbnail" in result:
+                result.pop("thumbnail")
+            print("")
+            print(n, result)
```

Comment:
I don't know if this script is meant to be used by someone else some day, but usually it is recommended to use `logging` package instead of `print`. Again, double check scripts links I gave you.